src/main/scala/com/campudus/tableaux/arguments.scala

```diff
@@ -33,12 +33,14 @@ case class FailArg[A](ex: CustomException) extends ArgumentCheck[A] {
   */
 object ArgumentChecker {
 
-  def notNull[A](x: => A, name: String): ArgumentCheck[A] = try {
-    if (x != null) OkArg(x.asInstanceOf[A]) else FailArg(InvalidJsonException(s"Warning: $name is null", "null"))
-  } catch {
-    case npe: NullPointerException => FailArg(InvalidJsonException(s"Warning: $name is null", "null"))
-    case cce: ClassCastException => FailArg(InvalidJsonException(s"Warning: $name should be another type. Error: ${cce.getMessage}", "invalid"))
-  }
+  def notNull[A](x: => A, name: String): ArgumentCheck[A] =
+    try {
+      if (x != null) OkArg(x.asInstanceOf[A]) else FailArg(InvalidJsonException(s"Warning: $name is null", "null"))
```

Comment:
Would be nice to see what happens if maximum chars per line is reached here (I hope it would change to:
```scala
if (something) {
  thisGetsReturned
} else {
  thatGetsReturned
}
```